### PR TITLE
Added 2-D array in c and c++

### LINKED ISSUE #1686: Add 2-D Array in C/C++
I want to create a new folder namely 2-D array using C and C++
1. Addition of two matrices
2. Difference of two matrices
3. Multiplication of two matrices
4.Individual row sum
5.Individual column sum
6.Diagonal sum
etc...

### PR DESCRIPTION
**Pull request by shubhigupta991**
Fixes #1686 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- c
- c++

#### Files Added:
- 
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by jainaman224** [comment]
Add issue number.

**Comment by jainaman224** [comment]
Fix PR title.

File: 2_D_Array/C++/addition.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        cout<<"Enter 2nd array :"<<endl;
+        for(int i=0; i<m; i++)
+        {
+            for(int j=0; j<n; j++)
+            {
+                cin>>b[i][j];
+            }
+        }
+        int c[m][n];
+	//calculating sum of a and b matrices.
```
Comment (jainaman224) [inline 2_D_Array/C++/addition.cpp]: Fix indentation throughout the file.

File: 2_D_Array/C++/addition.cpp [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+int main()
+{
+    int m,n,k,l;
+    cout << "Enter size of 1st array :" << endl;
+    cout<<"Row :";
+    cin>>m;
+    cout<<"Column :";
+    cin>>n;
+    cout<< "Enter size of 2st array :" << endl;
```
Comment (jainaman224) [inline 2_D_Array/C++/addition.cpp]: 2st should be 2nd.

File: 2_D_Array/C++/addition.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+
+using namespace std;
+
```
Comment (jainaman224) [inline 2_D_Array/C++/addition.cpp]: Spaces around operators like `==`, `=`, `+`, `<<`, `>>` etc improves readability.

File: 2_D_Array/C++/addition.cpp [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+8
+9
+10
+11
+12
+Added Array :
+8       10      12
+14      16      18
+
+*/
```
Comment (jainaman224) [inline 2_D_Array/C++/addition.cpp]: Add a single new line at the end of file.

File: 2_D_Array/C++/diagonal sum.cpp [hunk 5]
```diff
@@ -0,0 +1,64 @@
+#include <iostream>
```
Comment (jainaman224) [inline 2_D_Array/C++/diagonal sum.cpp]: Fix file name.

**Comment by jainaman224** [comment]
@shubhigupta991 Are you working on it?

**Comment by shubhigupta991** [comment]
> @shubhigupta991 Are you working on it?

yes sir m working on it

**Comment by shubhigupta991** [comment]
i have made all the suggested changes please review. @jainaman224

File: 2_D_Array/C++/addition.cpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
+#include <iostream>
```
Comment (somya-kapoor) [inline 2_D_Array/C++/addition.cpp]: Please change the file name to  2-D_Array/C++/Addition.cpp

File: 2_D_Array/C++/addition.cpp [hunk 7]
```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+
+using namespace std;
+
+int main()
+{
+    int m,n,k,l;
```
Comment (somya-kapoor) [inline 2_D_Array/C++/addition.cpp]: Please add space after comma

File: 2_D_Array/C++/addition.cpp [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        //for storing first matrix.
+        cout << "Enter 1st array :"<<endl;
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cin >> a[i][j];
+            }
+        }
+	//for storing second matrix.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/addition.cpp]: Indentation of comment

File: 2_D_Array/C++/addition.cpp [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+	//calculating sum of a and b matrices.
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                c[i][j] = a[i][j] + b[i][j];
+            }
+        }
+	//for displaying output matrix.
+        cout<<"Added Array :"<<endl;
```
Comment (somya-kapoor) [inline 2_D_Array/C++/addition.cpp]: Add space around the operators.

File: 2_D_Array/C++/addition.cpp [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cout << c[i][j]<<"\t";
```
Comment (somya-kapoor) [inline 2_D_Array/C++/addition.cpp]: Add space around the operators.

File: 2_D_Array/C++/difference.cpp [hunk 11]
```diff
@@ -0,0 +1,95 @@
+#include <iostream>
```
Comment (somya-kapoor) [inline 2_D_Array/C++/difference.cpp]: Please change the file name to 2-D_Array/C++/Difference.cpp

File: 2_D_Array/C++/difference.cpp [hunk 12]
```diff
@@ -0,0 +1,95 @@
+#include <iostream>
+
+using namespace std;
+
+int main()
+{
+    int m,n,k,l;
```
Comment (somya-kapoor) [inline 2_D_Array/C++/difference.cpp]: Please add space after comma

File: 2_D_Array/C++/difference.cpp [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+    cout << "Row :";
+    cin >> k;
+    cout << "Column :";
+    cin >> l;
+    int a[m][n], b[k][l];
+    if(m == k && n == l)
+    {
+        //for storing first matrix.
+        cout << "Enter 1st array :" << endl;
+        for(int i=0; i<m; i++)
```
Comment (somya-kapoor) [inline 2_D_Array/C++/difference.cpp]: Add space around the operator

File: 2_D_Array/C++/difference.cpp [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+        {
+            for(int j=0; j<n; j++)
```
Comment (somya-kapoor) [inline 2_D_Array/C++/difference.cpp]: Add space around the operator

File: 2_D_Array/C++/difference.cpp [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+        cout << "Enter 2nd array :" << endl;
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cin >> b[i][j];
+            }
+        }
+        int c[m][n];
+	//calculating subtraction of a and b matrices.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/difference.cpp]: Indentation

**Comment by somya-kapoor** [comment]
@shubhigupta991 Please do the required changes in all files.

**Comment by shubhigupta991** [comment]
I have made changes @somya-kapoor

File: 2_D_Array/C++/Addition.cpp [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+    cin >> n;
+    cout << "Enter size of 2nd array :" << endl;
+    cout << "Row :";
+    cin >> k;
+    cout << "Column :";
+    cin >> l;
+    int a[m][n], b[k][l];
+    if(m == k && n == l)
+    {
+        cout << "Enter 1st array :"<<endl;				//for storing first matrix.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Addition.cpp]: Add space around <<

File: 2_D_Array/C++/Addition.cpp [hunk 17]
```diff
@@ -0,0 +1,90 @@
+#include <iostream>
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Addition.cpp]: Add comments above the statement in the whole files.

File: 2_D_Array/C++/Sum of both diagonals.cpp [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+    int n,m;
+    cout << "Enter size of matrix :"<< endl;
+    cout << "Row : ";
+    cin >> m;
+    cout << "Column : ";
+    cin >> n;
+    int a[m][n];
+    if(m == n)									//for storing matrix.
+    {
+        cout << "Enter Matrix :"<<endl;
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Sum of both diagonals.cpp]: Add space around <<

File: 2_D_Array/C++/Sum of both diagonals.cpp [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cin >> a[i][j];
+            }
+        }
+        int sum = 0;			
+        for(int i = 0; i < m; i++)						//for calculating sum both the diagonals.
+        {
+            for(int j = 0; j <   n; j++)
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Sum of both diagonals.cpp]: Remove the space. it should be j < n

File: 2_D_Array/C/Addition.c [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+    scanf("%d",&l);
+    int a[m][n], b[k][l];
+    if(m == k && n == l)
+    {
+        printf("Enter 1st array : \n");						//for storing first matrix.
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                scanf("%d",&a[i][j]);
```
Comment (somya-kapoor) [inline 2_D_Array/C/Addition.c]: Add a space after ,

File: 2_D_Array/C/Addition.c [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+            }
+        }
+        printf("Enter 2nd array :\n");						//for storing second matrix.
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                scanf("%d",&b[i][j]);
```
Comment (somya-kapoor) [inline 2_D_Array/C/Addition.c]: Add a space after , . Make changes to the whole file.

**Comment by shubhigupta991** [comment]
I have made the suggested changes @somya-kapoor

File: 2_D_Array/C++/Addition.cpp [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+    cout << "Enter size of 2nd array :" << endl;
+    cout << "Row :";
+    cin >> k;
+    cout << "Column :";
+    cin >> l;
+    int a[m][n], b[k][l];
+    if(m == k && n == l)
+    {
+
+	//for storing first matrix.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Addition.cpp]: inconsistency in indentation

File: 2_D_Array/C++/Addition.cpp [hunk 23]
```diff
@@ -0,0 +1,99 @@
...
+
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Addition.cpp]: remove this line

File: 2_D_Array/C++/Addition.cpp [hunk 24]
```diff
@@ -0,0 +1,99 @@
...
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cin >> a[i][j];
+            }
+        }
+
+	 //for storing second matrix.
+
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Addition.cpp]: remove this line

File: 2_D_Array/C++/Addition.cpp [hunk 25]
```diff
@@ -0,0 +1,99 @@
...
+        cout << "Enter 1st array :" << endl;				
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cin >> a[i][j];
+            }
+        }
+
+	 //for storing second matrix.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Addition.cpp]: inconsistency in indentation

File: 2_D_Array/C++/Addition.cpp (hunk 17 above)
Comment (somya-kapoor) [reply 2_D_Array/C++/Addition.cpp]: Don't leave a line after the comments

File: 2_D_Array/C/Individual row sum.c [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            sum += a[i][j];
+        }
+        printf("\nSum of row %d", i+1);
+        printf(" : %d", sum);
+    }
+    printf("%d", "\n");
+    return 0;
+}
+/*
```
Comment (somya-kapoor) [inline 2_D_Array/C/Individual row sum.c]: leave a line above it

**Comment by somya-kapoor** [comment]
@shubhigupta991 Are you working on it?

**Comment by shubhigupta991** [comment]
> @shubhigupta991 Are you working on it?

yess

**Comment by shubhigupta991** [comment]
@somya-kapoor please review changes

File: 2_D_Array/C++/Addition.cpp (hunk 22 above)
Comment (somya-kapoor) [reply 2_D_Array/C++/Addition.cpp:22]: inconsistency in indentation

File: 2_D_Array/C++/Addition.cpp [hunk 27]
```diff
@@ -0,0 +1,95 @@
...
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cin >> b[i][j];
+            }
+        }
+        int c[m][n];
+
+	 //calculating sum of a and b matrices.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Addition.cpp:43]: inconsistency in indentation

File: 2_D_Array/C++/Addition.cpp [hunk 28]
```diff
@@ -0,0 +1,95 @@
...
+        for(int i = 0; i < m; i++)                                     
+        {
+            for(int j = 0; j < n; j++)
+            {
+                c[i][j] = a[i][j] + b[i][j];
+            }
+        }
+
+	//for displaying output matrix.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Addition.cpp:52]: inconsistency in indentation

File: 2_D_Array/C++/Difference.cpp [hunk 29]
```diff
@@ -0,0 +1,96 @@
...
+    cout << "Enter size of 2nd array :" << endl;
+    cout << "Row :";
+    cin >> k;
+    cout << "Column :";
+    cin >> l;
+    int a[m][n], b[k][l];
+    if(m == k && n == l)
+    {
+
+	//for storing first matrix.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Difference.cpp:22]: inconsistency in indentation

File: 2_D_Array/C++/Difference.cpp [hunk 30]
```diff
@@ -0,0 +1,96 @@
...
+        cout << "Enter 1st array :" << endl;					
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cin >> a[i][j];
+            }
+        }
+
+	//for storing second matrix.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Difference.cpp:32]: inconsistency in indentation

File: 2_D_Array/C++/Difference.cpp [hunk 31]
```diff
@@ -0,0 +1,96 @@
...
+        for(int i = 0; i < m; i++)
+        {
+            for(int j = 0; j < n; j++)
+            {
+                cin >> b[i][j];
+            }
+        }
+        int c[m][n];
+
+	//calculating subtraction of a and b matrices.
```
Comment (somya-kapoor) [inline 2_D_Array/C++/Difference.cpp:43]: inconsistency in indentation. correct this in whole files.

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this issue

**Pull request closed without merging**